### PR TITLE
Strip optional '.git' suffix from https server remote name.

### PR DESCRIPTION
**Pull request by salexander6**
Fixes an error when creating a PR, similar to what is described here https://github.com/peter-evans/create-pull-request/issues/1096

My repository name was identified as `data.git` and the `.git` suffix caused an error when creating the PR here https://github.com/peter-evans/create-pull-request/blob/main/src/github-helper.ts#L47

```
Determining the base and head repositories
  /usr/bin/git config --local --get remote.origin.url
  https://***@github.com/99spokes/data.git
```

```
Create or update the pull request
  Attempting creation of pull request {
    "owner": "99spokes",
    "repo": "data.git",
    "title": "Update Products",
    "head": "99spokes:update-products",
    "base": "main",
    "body": "Automated changes by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action",
    "draft": false
  }
  Error: Not Found
```

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -46,12 +46,14 @@ export function getRemoteDetail(remoteUrl: string): RemoteDetail {
     throw new Error('Could not parse GitHub Server name')
   }
 
+  remoteUrl = remoteUrl.replace(/\.git$/, '')
+
   const httpsUrlPattern = new RegExp(
     '^https?://.*@?' + githubServerMatch[1] + '/(.+/.+)$',
```
Comment (peter-evans) [inline src/utils.ts:52]: I would prefer to fix this by only changing the regex.
```typescript
'^https?://.*@?' + githubServerMatch[1] + '/(.+/.+?)(.git)?$',
```

Also, please add a test for this pattern here:
https://github.com/peter-evans/create-pull-request/blob/main/__test__/utils.unit.test.ts#L35-L53

**Review by peter-evans — CHANGES_REQUESTED** [review]

File: src/utils.ts (hunk 1 above)
Comment (salexander6) [reply src/utils.ts:52]: > I would prefer to fix this by only changing the regex.
👍

**Comment by salexander6** [comment]
> Please could you show me a workflow that results in this happening. Perhaps you are not using the official `actions/checkout` action to checkout the repository?

The repo is private but here's the relevant workflow section

```
steps:
  - name: Check out 99spokes/data
    run: |
      git clone --filter=blob:none --no-checkout https://$BOT_GITHUB_TOKEN@github.com/99spokes/data.git
      cd data
      git sparse-checkout init --cone --sparse-index
      git sparse-checkout set products
      git checkout main
    env:
      BOT_GITHUB_TOKEN: ${{ secrets.BOT_GITHUB_TOKEN }}

  - name: Download artifacts
    uses: actions/download-artifact@v2
    with:
      name: products
      path: data/products

  - name: Create Pull Request
    uses: 99spokes/create-pull-request@fix-https-repository-name
    with:
      path: data
      token: ${{ secrets.BOT_GITHUB_TOKEN }}
      branch: update-products
      commit-message: Update products.
      title: 'Update Products'
      labels: automerge
   ```   

Adding this here has me wondering if cloning without the `.git` extension would work around the issue 🤔

**Comment by peter-evans** [comment]
Ok, it makes sense now. Not using the official checkout action causes the remote URL to include the `.git` suffix. It's an easy fix though, so I'm happy to get this merged, regardless of whether or not you are able to work around it.

**Review by peter-evans — APPROVED** [review]

**Comment by peter-evans** [comment]
Thank you!

Released as `v4.0.1` / `v4`

**Pull request merged**